sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CSVCarbonWriter.java

```diff
@@ -72,6 +88,75 @@ public void write(Object object) throws IOException {
     }
   }
 
+  private CsvParser buildCsvParser(Configuration conf) {
+    CsvParserSettings settings = CSVInputFormat.extractCsvParserSettings(conf);
+    return new CsvParser(settings);
+  }
+
+  @Override
+  public void validateAndSetDataFiles(CarbonFile[] dataFiles) throws IOException {
```

Comment:
check if this can be private

Reply:
I have changed the method name as `setDataFiles` but we have to expose this as public so that all the writer can use from `CarbonWriterBuilder`. Also, I have put this in carbonWriter because it will clean the code at the time of building the carbonWriter.